podpac/core/interpolation/interpolator.py

```diff
@@ -293,20 +293,40 @@ def _loop_helper(
                     else:
                         src_idx = None  # There is no closest neighbor within the tolerance
                         continue
+
                 else:
                     src_idx = idx
 
                 output_data.loc[idx] = self._loop_helper(
                     func,
                     keep_dims,
                     udims,
-                    source_coordinates,
+                    source_coordinates.drop(dim),
                     source_data.loc[src_idx],
-                    eval_coordinates,
+                    eval_coordinates.drop(dim),
```

Comment:
@mlshapiro This was the other change that I had to make for heterogeneous interpolation so that source_coordinates matched the source_data, and the eval_coordinates matched the output_data.

Reply:
@jmilloy yes i'll review this branch today

Reply:
this makes a lot of sense. The loop_helper in general is a bit fragile since it only supports lat/lon grids interpolating over time/alt. This sees to work in my testing.